internal/security/secretstore/init.go

```diff
@@ -109,19 +129,36 @@ func (b *Bootstrap) BootstrapHandler(ctx context.Context, _ *sync.WaitGroup, _ s
 					initResponse.RootToken = ""
 					lc.Info("Root token stripped from init response for security reasons")
 				}
-				if err := saveInitResponse(lc, fileOpener, configuration.SecretService, &initResponse); err != nil {
-					lc.Error(fmt.Sprintf("unable to save init response: %s", err.Error()))
-					os.Exit(1)
-				}
 				_, err = vc.Unseal(&initResponse)
 				if err == nil {
 					shouldContinue = false
 				}
+				// We need the unencrypted initResponse in order to generate a temporary root token later
+				// Make a copy and save the copy, possibly encrypted
+				encryptedInitResponse := initResponse
+				// Optionally encrypt the vault init response based on whether encryption was enabled
+				if vmkEncryption.IsEncrypting() {
+					if err := vmkEncryption.EncryptInitResponse(&encryptedInitResponse); err != nil {
+						lc.Error(fmt.Sprintf("failed to encrypt init response from secret store: %s", err.Error()))
+						os.Exit(1) // can't return false since in anonymous func()
+					}
+				}
+				if err := saveInitResponse(lc, fileOpener, configuration.SecretService, &encryptedInitResponse); err != nil {
+					lc.Error(fmt.Sprintf("unable to save init response: %s", err.Error()))
+					os.Exit(1) // can't return false since in anonymous func()
+				}
 			case http.StatusServiceUnavailable:
 				lc.Info(fmt.Sprintf("vault is sealed (status code: %d). Starting unseal phase", sCode))
 				if err := loadInitResponse(lc, fileOpener, configuration.SecretService, &initResponse); err != nil {
 					lc.Error(fmt.Sprintf("unable to load init response: %s", err.Error()))
-					os.Exit(1)
+					os.Exit(1) // can't return false since in anonymous func()
+				}
+				// Optionally decrypt the vault init response based on whether encryption was enabled
+				if vmkEncryption.IsEncrypting() {
+					if err := vmkEncryption.DecryptInitResponse(&initResponse); err != nil {
+						lc.Error(fmt.Sprintf("failed to decrypt key shares for sercret store unsealing: %s", err.Error()))
+						os.Exit(1) // can't return false since in anonymous func()
```

Comment:
anonymous func() can have a `bool` return type and then check it on line 138 and return `false` if it is `false`

Reply:
can do that.